xarray/core/variable.py

```diff
@@ -406,6 +407,96 @@ def __getitem__(self, key):
         return type(self)(dims, values, self._attrs, self._encoding,
                           fastpath=True)
 
+    def _broadcast_indexes(self, key):
+        """
+        Parameters
+        -----------
+        key: One of
+            array
+            a mapping of dimension names to index.
+
+        Returns
+        -------
+        dims: Tuple of strings.
+            Dimension of the resultant variable.
+        indexers: list of integer, array-like, or slice. This is aligned
+            along self.dims.
+        """
+        if not utils.is_dict_like(key):
+            key = {self.dims[0]: key}
+        example_v = None
+        indexes = OrderedDict()
+        for k, v in key.items():
+            if not isinstance(v, (integer_types, slice, Variable)):
+                if not hasattr(key, 'ndim'):  # convert list or tuple
+                    v = np.array(v)
+            if example_v is None and isinstance(v, Variable):
+                example_v = v
+            indexes[k] = v
+
+        # When all the keys are array or integer, slice
```

Comment:
@shoyer
> We need this path because "basic indexing" is much faster than "advanced indexing" in NumPy, as it avoids copying data.

Thanks for the suggestion!
And also thanks for the reference implementation.
(What a simple logic!!)

Please let me make sure one thing.

What would be expected by the following?
```python
v = Variable(['x', 'y'], [[0, 1, 2], [3, 4, 5]])
ind_x = Variable(['a'], [0, 1])
v.getitem2(dict(x=ind_x, y=[1, 0])))
```
It should be understood as `y=Variable(['y'], [1, 0])`?
I thought `y=[0, 1]` would be `y=Variable(['a'], [1, 0])`...